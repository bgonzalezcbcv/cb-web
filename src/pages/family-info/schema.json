{
	"type": "object",
	"title": "Editar responsable",
	"properties": {
		"role": {
			"oneOf": [
				{
					"const": "mother",
					"title": "Madre"
				},
				{
					"const": "father",
					"title": "Padre"
				},
				{
					"const": "guardian",
					"title": "Tutor"
				}
			],
			"type": "string",
			"title": "Tipo de responsable",
			"default": "mother"
		},
		"fullName": {
			"type": "string",
			"title": "Nombre completo"
		},
		"birthDate": {
			"type": "string",
			"format": "date",
			"dateTimeFormat": "DD/MM/YY",
			"dateTimeSaveFormat": "DD-MM-YY",
			"title": "Fecha de nacimiento"
		},
		"birthCountry": {
			"type": "string",
			"title": "País de nacimiento"
		},
		"nationality": {
			"type": "string",
			"title": "Nacionalidad"
		},
		"motherToungue": {
			"type": "string",
			"title": "Lengua materna"
		},
		"ci": {
			"type": "string",
			"title": "Cedula de identidad",
			"pattern": "\\d(\\.\\d{3}){2}-\\d"
		},
		"maritalStatus": {
			"oneOf": [
				{
					"const": "single",
					"title": "Soltero/a"
				},
				{
					"const": "married",
					"title": "Casado/a"
				},
				{
					"const": "widow",
					"title": "Viudo/a"
				}
			],
			"type": "string",
			"title": "Estado civil",
			"default": "single"
		},
		"cellphone": {
			"type": "string",
			"title": "Teléfono celular",
			"pattern": "\\d{9}"
		},
		"email": {
			"type": "string",
			"title": "E-mail"
		},
		"address": {
			"type": "string",
			"title": "Dirección"
		},
		"residenceNeighborhood": {
			"type": "string",
			"title": "Barrio de residencia"
		},
		"educationLevel": {
			"oneOf": [
				{
					"const": "none",
					"title": "Ninguno"
				},
				{
					"const": "school",
					"title": "Primaria"
				},
				{
					"const": "highSchool",
					"title": "Secundaria"
				},
				{
					"const": "tertiary",
					"title": "Terciario"
				}
			],
			"title": "Nivel educativo",
			"default": "none"
		},
		"occupation": {
			"type": "string",
			"title": "Trabajo-Ocupación"
		},
		"workplace": {
			"type": "string",
			"title": "Empresa de trabajo"
		},
		"workplaceAddress": {
			"type": "string",
			"title": "Dirección laboral"
		},
		"workplaceNeighborhood": {
			"type": "string",
			"title": "Barrio laboral"
		},
		"workplacePhone": {
			"type": "number",
			"title": "Teléfono laboral",
			"pattern": "\\d{8,9}"
		}
	},
	"required": ["type", "ci"]
}
