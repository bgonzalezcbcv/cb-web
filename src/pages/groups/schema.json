{
  "type": "object",
  "properties": {
	"cycle": {
	  "oneOf": [
		{
		  "const": "maternal",
		  "title": "Maternal"
		},
		{
		  "const": "inicial",
		  "title": "Inicial"
		},
		{
		  "const": "primaria",
		  "title": "Primaria"
		},
		{
		  "const": "secundaria",
		  "title": "Secundaria"
		}
	  ],
	  "type": "string",
	  "title": "Ciclo",
	  "errorMessage": "Debe seleccionar un ciclo"
	},
	"class": {
	  "type": "string",
	  "title": "Clase",
	  "minLength": 1,
	  "errorMessage": "Este campo es requerido."
	},
	"subgroup": {
	  "type": "string",
	  "title": "Subgrupo",
	  "minLength": 1,
	  "errorMessage": "Este campo es requerido."
	},
	"year": {
	  "type": "integer",
	  "title": "AÃ±o",
	  "minLength": 1,
	  "errorMessage": "Este campo es requerido."
	}
  }
}