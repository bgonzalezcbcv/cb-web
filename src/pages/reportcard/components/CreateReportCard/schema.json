{
	"type": "object",
	"properties": {
		"start_period": {
			"type": "string",
			"title": "Comienzo del período",
			"minLength": 1,
			"errorMessage": "Debe ser una fecha válida."
		},
		"end_period": {
			"type": "string",
			"title": "Fin del período",
			"minLength": 1,
			"errorMessage": "Debe ser una fecha válida."
		}
	},
	"required": ["start_period", "end_period"]
}
