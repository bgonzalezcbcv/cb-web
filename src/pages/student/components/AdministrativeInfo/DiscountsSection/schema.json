{
	"type": "object",
	"title": "Descuento",
	"properties": {
		"starting_date": {
			"title": "Comienzo",
			"type": "string"
		},
		"ending_date": {
			"title": "Fin",
			"type": "string"
		},
		"explanation": {
			"oneOf": [
				{
					"const": "hermano",
					"title": "Hermano"
				},
				{
					"const": "resolución",
					"title": "Resolución"
				}
			],
			"title": "Explicación",
			"type": "string",
			"default": "Sibling"
		},
		"percentage": {
			"title": "%",
			"type": "integer",
			"minimum": 0,
			"maximum": 100
		},
		"type": {
			"oneOf": [
				{
					"const": "dirección",
					"title": "Dirección"
				},
				{
					"const": "asistente social",
					"title": "Asistente Social"
				}
			],
			"title": "Tipo",
			"type": "string",
			"default": "Direction"
		},
		"report_url": {
			"title": "Informe administrativo",
			"type": "string"
		},
		"description": {
			"title": "Descripción",
			"type": "string"
		},
		"resolution_url": {
			"title": "Resolución",
			"type": "string"
		}
	},
	"required": ["starting_date", "ending_date", "explanation", "percentage", "type", "report_url"]
}
