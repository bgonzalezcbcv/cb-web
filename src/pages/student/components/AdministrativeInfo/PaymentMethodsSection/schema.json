{
	"type": "object",
	"title": "Descuento",
	"properties": {
		"method": {
			"oneOf": [
				{
					"const": "contado",
					"title": "Contado"
				},
				{
					"const": "financiacion",
					"title": "Financiación"
				},
				{
					"const": "licitacion",
					"title": "Licitación"
				}
			],
			"title": "Método de pago",
			"type": "string"
		},
		"year": {
			"title": "Año",
			"type": "integer",
			"pattern": "\\d{4}"
		},
		"yearly_payment_url": {
			"title": "Pago anualidad firmado",
			"type": "string"
		}
	},
	"required": ["method", "year"]
}
