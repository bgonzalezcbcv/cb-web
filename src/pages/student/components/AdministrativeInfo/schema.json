{
	"type": "object",
	"title": "Informaci贸n administrativa",
	"properties": {
		"enrollmentDate": {
			"type": "string",
			"format": "date",
			"title": "Fecha de inscripci贸n"
		},
		"startDate": {
			"type": "string",
			"format": "date",
			"title": "Fecha de comienzo"
		},
		"scholarshipType": {
			"type": "string",
			"oneOf": [
				{
					"const": "licitaci贸n",
					"title": "Licitaci贸n"
				},
				{
					"const": "bonificada",
					"title": "Bonificada"
				},
				{
					"const": "convenio",
					"title": "Convenio"
				},
				{
					"const": "especial",
					"title": "Especial"
				}
			],
			"title": "Tipo de escolaridad",
			"default": "none"
		}
	}
}
