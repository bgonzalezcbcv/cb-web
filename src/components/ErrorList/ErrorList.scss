.error-list {
	width: max-content;

	ul {
		list-style-type: none;
		margin-top: 0;
	}
}
