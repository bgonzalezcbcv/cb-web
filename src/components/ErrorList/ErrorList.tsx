import _ from "lodash";
import React from "react";

import { getTitleFromSchema } from "../../core/CoreHelper";
import { JsonSchema7 } from "@jsonforms/core";
import { Box, List, ListItem, Typography } from "@mui/material";

import "./ErrorList.scss";

interface ErrorRow {
	title: string;
	message: string;
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
type ErrorsEntry = Record<string, ErrorRow | ErrorsEntry>;

interface ErrorListProps {
	name?: string;
	path: string;
	value: ErrorsEntry;
	schema: JsonSchema7;
}

function ErrorList(props: ErrorListProps): JSX.Element {
	const { name, path, value, schema } = props;

	function printRow(name: string, message: string): React.ReactElement {
		return (
			<ListItem key={`error-row-${name}`}>
				<Box display="flex" flexDirection="row" gap="5px">
					<Typography fontWeight="bold" color="darkred">{`${name}: `}</Typography>
					<Typography>{`${message}`}</Typography>
				</Box>
			</ListItem>
		);
	}

	function printObject(name: string, obj: Record<string, unknown>): React.ReactElement {
		return (
			<ListItem className="error-list">
				<Box>
					{name ? (
						<Typography fontWeight="bold" color="darkred" fontSize="medium">
							{name}
						</Typography>
					) : null}

					<List>
						{Object.entries(obj).map((entry, index) => (
							<ErrorList
								key={`error-row-obj-${index}`} //
								path={path !== "" ? path + "." + entry[0] : entry[0]}
								value={entry[1]}
								schema={schema}
							/>
						))}
					</List>
				</Box>
			</ListItem>
		);
	}

	function printArray(name: string, arr: Record<string, unknown>[]): React.ReactElement {
		return (
			<ListItem>
				<Box>
					<Typography fontWeight="bold" color="darkred" fontSize="medium">
						{name}
					</Typography>

					<List>
						{arr.map((entry, index) => (
							<ErrorList
								key={`error-row-obj-${index}`} //
								path={path + `[${index}]`}
								name={`${index + 1} Â°`}
								value={entry}
								schema={schema}
							/>
						))}
					</List>
				</Box>
			</ListItem>
		);
	}

	if (value.message) return printRow(name ?? value.title, value.message);

	if (_.isArray(value)) return printArray(name ?? getTitleFromSchema(path, schema) ?? "", value);

	if (_.isPlainObject(value)) return printObject(name ?? getTitleFromSchema(path, schema) ?? "", value as Record<string, unknown>);

	return <></>;
}

export default ErrorList;
