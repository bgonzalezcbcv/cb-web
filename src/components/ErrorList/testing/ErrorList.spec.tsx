import React from "react";
import { render } from "@testing-library/react";

import ErrorList from "../ErrorList";

import data from "./testingData.json";
import schema from "./testingDataSchema.json";

describe("ErrorList", () => {
	it("should render a correct list", () => {
		const wrapper = render(<ErrorList path="" schema={schema} value={data} />);

		expect(wrapper).toMatchSnapshot();
	});
});
