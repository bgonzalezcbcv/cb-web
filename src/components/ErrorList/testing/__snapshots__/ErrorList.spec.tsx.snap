// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorList should render a correct list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiBox-root css-tx8ton"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                >
                  testTitle: 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  testMessage
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiBox-root css-tx8ton"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                >
                  testTitle: 
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  testMessage
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                >
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiBox-root css-tx8ton"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                      >
                        testTitle: 
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        testMessage
                      </p>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiBox-root css-tx8ton"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                      >
                        testTitle: 
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        testMessage
                      </p>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                      />
                      <ul
                        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                      >
                        <li
                          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
                        >
                          <div
                            class="MuiBox-root css-0"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                            >
                              1 °
                            </p>
                            <ul
                              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                            >
                              <li
                                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                              >
                                <div
                                  class="MuiBox-root css-tx8ton"
                                >
                                  <p
                                    class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                                  >
                                    testTitle: 
                                  </p>
                                  <p
                                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                                  >
                                    testMessage
                                  </p>
                                </div>
                              </li>
                            </ul>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
                        >
                          <div
                            class="MuiBox-root css-0"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                            >
                              2 °
                            </p>
                            <ul
                              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                            >
                              <li
                                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                              >
                                <div
                                  class="MuiBox-root css-tx8ton"
                                >
                                  <p
                                    class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                                  >
                                    testTitle: 
                                  </p>
                                  <p
                                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                                  >
                                    testMessage
                                  </p>
                                </div>
                              </li>
                            </ul>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiBox-root css-tx8ton"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
              >
                testTitle: 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                testMessage
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiBox-root css-tx8ton"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
              >
                testTitle: 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                testMessage
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <ul
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                >
                  <div
                    class="MuiBox-root css-tx8ton"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                    >
                      testTitle: 
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      testMessage
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                >
                  <div
                    class="MuiBox-root css-tx8ton"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                    >
                      testTitle: 
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      testMessage
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                    />
                    <ul
                      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                    >
                      <li
                        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                          >
                            1 °
                          </p>
                          <ul
                            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                          >
                            <li
                              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                            >
                              <div
                                class="MuiBox-root css-tx8ton"
                              >
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                                >
                                  testTitle: 
                                </p>
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                                >
                                  testMessage
                                </p>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </li>
                      <li
                        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding error-list css-1p823my-MuiListItem-root"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-yniyb8-MuiTypography-root"
                          >
                            2 °
                          </p>
                          <ul
                            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                          >
                            <li
                              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                            >
                              <div
                                class="MuiBox-root css-tx8ton"
                              >
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-uzsolv-MuiTypography-root"
                                >
                                  testTitle: 
                                </p>
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                                >
                                  testMessage
                                </p>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
